File: Queue/java/Queue.java
class node
{
    public int data;
    public node next;
}

class QueueX
{
    private node first;
    private int iCount;

    public QueueX()
    {
        this.first = null;
        this.iCount = 0;
    }

    void enqueue (int iNo)  // insert last
    {
        node newn = new node();

        newn.data = iNo;
        newn.next = null;

        if(first == null)
        {
            first = newn;
        }
        else
        {
            node temp = first;

            while(temp.next != null)
            {
                temp=temp.next;

               
            }
            temp.next = newn;
        }
        iCount++;

    }

    int dequeue()     // delete first
    {
        node temp = first;
        if(first ==null)
        {
            System.out.println("Queue is empty :");
            return -1;
        }
        else
        {
            first = first.next;
            iCount--;
            return temp.data;
        }

       
    }

    void Display()
    {
        node temp = first;
        while(temp != null)
        {
            System.out.print(" | "+temp.data+" | - ");
            temp = temp.next ;
        }

    }

    int Count ()
    {
        return iCount;
    }

}

class Queue
{
    public static void main(String a[])
    {
        QueueX sobj = new  QueueX ();
        int iRet = 0;

        sobj. enqueue (10);
        sobj. enqueue (20);
        sobj. enqueue (30);
        sobj.enqueue(40);

        sobj.Display();
        iRet = sobj.Count();
        System.out.println("element in queue:"+iRet);

        sobj.dequeue();
        
        sobj.Display();
        iRet = sobj.Count();
        System.out.println("element in queue:"+iRet);

    }
}  

File: Queue/c/Queue.c
#include<stdio.h>
#include <stdlib.h>

struct node
{
    int data;
    struct node* next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node ** PPNODE;

void Enqueue (PPNODE first,int ino ) // insert last
{
    PNODE newn = (PNODE *)malloc (sizeof(NODE));
    PNODE temp = NULL;

    newn->data = ino;
    newn->next = NULL;

    if( *first == NULL)
    {
        *first = newn;
    }
    else
    {
        temp = *first;

        while(temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newn;
    }
}

int Dequeue (PPNODE first )   // delete first
{
    PNODE temp = NULL;
    int data = 0;

    if(*first == NULL)
    {
        printf("queue is empty \n");
        return -1;
    }
    else
    {
        temp = *first ;
        *first = temp->next;
        data = temp->data;
        free(temp);

        return data;
    }
    
}

int Count(PNODE first)
{
    PNODE temp = first;
    int iCount = 0;

    while(temp != NULL)
    {
        iCount ++;
        temp = temp->next;
    }

    return iCount;

}

void Display(PNODE first)
{
    PNODE temp = first;

    while(temp != NULL)
    {
        printf(" |  %d   | - " ,temp->data);
        temp = temp-> next;
    }
    printf("\n");


}

int main()
{
    PNODE head = NULL;
    int iRet =0;

    Enqueue(&head,10);
    Enqueue(&head,20);
    Enqueue(&head,30);
    Enqueue(&head,40);

    Display(head);

    iRet = Count(head);
    printf("no of element in queue :%d\n",iRet);

    Dequeue(&head);

    Display(head);

    iRet = Count(head);
    printf("no of element in queue :%d\n",iRet);

    return 0;
}